.buildkite/pipeline_cpu_template.py

```diff
@@ -0,0 +1,72 @@
+#!/usr/bin/env python3
+# Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# SPDX-License-Identifier: Apache-2.0
+
+"""Generate Buildkite CPU template pipelines dynamically"""
+
+import argparse
+import json
+
+from common import group, DEFAULT_INSTANCES, DEFAULT_KERNELS
+
+cpu_template_test = {
+    "rdmsr": [
+        {
+            "label": "🖴 rdmsr",
+            "test_path": "integration_tests/functional/test_cpu_features.py::test_cpu_rdmsr",
+            "devtool_opts": " ",
+            "instances": ["m5d.metal", "m6i.metal", "m6a.metal"],
+        },
+    ],
+}
+
+
+def build_group(test):
+    """Build a Buildkite pipeline `group` step"""
+    devtool_opts = test.pop("devtool_opts")
+    test_path = test.pop("test_path")
+    return group(
+        label=test.pop("label"),
+        command=f"./tools/devtool -y test {devtool_opts} -- --nonci -s --dump-results-to-file --log-cli-level=INFO {test_path}",
+        instances=test.pop("instances"),
+        kernels=test.pop("kernels"),
+        # and the rest can be command arguments
+        **test
+    )
+
+
+parser = argparse.ArgumentParser()
+parser.add_argument(
+    "--test",
+    required=True,
+    choices=list(cpu_template_test.keys()),
+    help="CPU template test"
+)
+parser.add_argument(
+    "--add-instance",
+    required=False,
+    action="append",
+    default=DEFAULT_INSTANCES,
+)
```

Comment:
since you already specify which instances you run on, you may not need this parameter here and below